systers_portal/meetup/tests/test_views.py

```diff
@@ -1868,18 +1899,88 @@ def test_api_for_vms_get(self):
+        self.meetup = Meetup.objects.create(title='Foo Bar Baz', slug='foo-bar-baz',
+                                            date='2018-06-16',
+                                            time=timezone.now().time(),
+                                            description='This is test Meetup',
+                                            venue='Foo Systers',
+                                            meetup_location=self.meetup_location,
+                                            created_by=self.systers_user,
+                                            last_updated=timezone.now())
+        self.meetup2 = Meetup.objects.create(title='Foo Baz', slug='foobar',
+                                             date='2018-06-12',
+                                             time=timezone.now().time(),
+                                             description='This is new test Meetup',
+                                             venue='Foo Systers',
+                                             meetup_location=self.meetup_location,
+                                             created_by=self.systers_user,
+                                             last_updated=timezone.now())
+        self.meetup3 = Meetup.objects.create(title='Foob Baz', slug='foobarbaz',
+                                             date='2018-06-13',
+                                             time=timezone.now().time(),
+                                             description='This is test Meetup',
+                                             venue='Foo Systers',
+                                             meetup_location=self.meetup_location1,
+                                             created_by=self.systers_user,
+                                             last_updated=timezone.now())
+
+    def test_post_view(self):
+        """Test post view for all search requests"""
+        url = reverse('search_meetups')
+        data = {'date': '2018-06-12', 'meeetup_location': 'Foo Systers', 'keyword': 'new'}
+        response = self.client.post(url, data, format='json')
+        self.assertEqual(json.loads(response.content.decode('utf-8')),
+                         {'search_results':
+                         [{'date': '2018-06-12',
+                           'meetup': 'Foo Baz',
+                           'location': 'Foo Systers',
+                           'location_slug': 'foo',
+                           'meetup_slug': 'foobar',
+                           'distance': '',
+                           'unit': ''}]})
+
+        data1 = {'date': '2018-06-13', 'meeetup_location': 'Foo Systers1', 'keyword': 'test',
+                 'filter': 2}
+        response = self.client.post(url, data1, format='json')
+        self.assertEqual(json.loads(response.content.decode('utf-8')),
+                         {'search_results':
+                         [{'date': '2018-06-13',
+                           'meetup': 'Foob Baz',
```

Comment:
More test cases have to be added into the tests to cover every condition in the search view. In the `UpcomingMeetupsSearchView` as seen [here](https://coveralls.io/builds/18363379/source?filename=systers_portal/meetup/views.py)  some test cases haven't been checked by this.